contracts/EscrowAndStaking/StakingThales.sol

```diff
@@ -599,12 +685,118 @@ contract StakingThales is IStakingThales, Initializable, ProxyOwned, ProxyReentr
+        }
+
+        if (msg.sender == exoticBonds) {
+            if (lastExoticMarketsUpdatePeriod[account] < periodsOfStaking) {
+                exoticMarketsVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount = 0;
+                exoticMarketsVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].period = periodsOfStaking;
+                lastExoticMarketsUpdatePeriod[account] = periodsOfStaking;
+            }
+            exoticMarketsVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount = exoticMarketsVolume[
+                account
+            ][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount.add(amount);
+        }
+
+        if (msg.sender == sportsAMM) {
+            if (lastSportsAMMUpdatePeriod[account] < periodsOfStaking) {
+                sportsAMMVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount = 0;
+                sportsAMMVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].period = periodsOfStaking;
+                lastSportsAMMUpdatePeriod[account] = periodsOfStaking;
+            }
+            sportsAMMVolume[account][periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)].amount = sportsAMMVolume[account][
+                periodsOfStaking.mod(AMM_EXTRA_REWARD_PERIODS)
+            ].amount.add(amount);
+        }
+
         emit AMMVolumeUpdated(account, amount);
     }
 
+    /// @notice Merge account to transfer all staking amounts to another account
+    /// @param destAccount to merge into
+    function mergeAccount(address destAccount) external notPaused {
+        require(destAccount != address(0) && destAccount != msg.sender, "Invalid address");
+        require(
+            _calculateAvailableRewardsToClaim(msg.sender) == 0 && _calculateAvailableRewardsToClaim(destAccount) == 0,
+            "Cannot merge, claim rewards on both accounts before merging"
+        );
+        require(
+            !unstaking[msg.sender] && !unstaking[destAccount],
+            "Cannot merge, cancel unstaking on both accounts before merging"
+        );
+
+        iEscrowThales.mergeAccount(msg.sender, destAccount);
+
+        _stakedBalances[destAccount] = _stakedBalances[destAccount].add(_stakedBalances[msg.sender]);
+        stakerLifetimeRewardsClaimed[destAccount] = stakerLifetimeRewardsClaimed[destAccount].add(
+            stakerLifetimeRewardsClaimed[msg.sender]
+        );
+        stakerFeesClaimed[destAccount] = stakerFeesClaimed[destAccount].add(stakerFeesClaimed[msg.sender]);
+
+        _lastRewardsClaimedPeriod[destAccount] = periodsOfStaking;
+        _lastStakingPeriod[destAccount] = periodsOfStaking;
+        lastAMMUpdatePeriod[destAccount] = periodsOfStaking;
+
+        for (uint i = 1; i <= AMM_EXTRA_REWARD_PERIODS; i++) {
+            uint stakerAMMVolumeIndex = periodsOfStaking.add(i).mod(AMM_EXTRA_REWARD_PERIODS);
+            uint stakerAMMVolumePeriod = periodsOfStaking.sub(AMM_EXTRA_REWARD_PERIODS.sub(i));
```

Comment:
initialize `stakerAMMVolumeIndex` and `stakerAMMVolumePeriod` before the `for()` cycle

Reply:
Updated.